components/StarRating.tsx

```diff
@@ -1,31 +1,172 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { FaStar } from "react-icons/fa";
 
-const StarRating = () => {
+interface StarRatingProps {
+  openedData: any;
+}
+
+const StarRating: React.FC<StarRatingProps> = ({ openedData }) => {
   const [rating, setRating] = useState<number | null>(null);
   const [hover, setHover] = useState<number | undefined>(undefined);
+  const [ratingChanged, setRatingChanged] = useState<boolean>(false);
+  const [called, setCalled] = useState<boolean>(true);
+  const [hasRated, setHasRated] = useState<boolean>(false);
+  const [ratingCount, setRatingCount] = useState<number>(0);
+  const [averageRating, setAverageRating] = useState<number>(0);
+
+  // check if the rating has been changed, if so, submit the rating
+  useEffect(() => {
+    if (ratingChanged) {
+      handleRatingSubmit();
+    }
+  }, [ratingChanged]);
+
+  // this useEffect is called when the popup is opened and never again, it renders the rating if the user has rated before
+  useEffect(() => {
+    if (called) {
+      handleGetSingleRating();
+      handleGetAllRatings();
+      setCalled(false);
+    }
+  }, [called]);
+
+  // get average ratings and rating count
+  const handleGetAllRatings = async () => {
+    try {
+      const data = { courseId: openedData.id };
+      const query = new URLSearchParams(data);
+      const response = await fetch(
+        `http://localhost:3000/api/ratings/GET_all?${query.toString()}`,
```

Comment:
Is hard coding the API endpoint what was done previously?

Would this need to be changed **if** we were to ever deploy this?
I'm not too sure of how an alternative would work either though.
https://supabase.com/blog/fetching-and-caching-supabase-data-in-next-js-server-components, I'm not sure how relevant something like this could be.

Reply:
That is hard-coded legacy code will need to be changed for deployment.